kolibri/core/apps.py

```diff
@@ -124,13 +119,29 @@ def check_redis_settings():
                     "Kolibri, the following is suggested: maxmemory-policy allkeys-lru"
                 )
 
-            # should only change something when `config_use_conf` is also `False`
-            if helper.changed:
-                helper.save()
-            elif maxmemory == 0 or maxmemory_policy == "noeviction":
+            # default setting is "0", or no limit
+            maxmemory = helper.get_maxmemory()
+            if config_maxmemory > 0 and maxmemory != config_maxmemory:
+                used_memory = helper.get_used_memory()
+                if maxmemory < used_memory:
+                    logger.warning(
+                        "Redis was using {used_memory} before setting `maxmemory` configuration of {config_memory}".format(
+                            used_memory=bytes_for_humans(used_memory),
+                            config_memory=bytes_for_humans(config_maxmemory),
+                        )
+                    )
+
+                helper.set_maxmemory(config_maxmemory)
```

Comment:
should not this be after an `else` clausule, so we don't set the new config_maxmemory if we know it's gonna cause problems because it's less than the current used memory?

Reply:
No it's fine because setting the policy first will not trigger the issue you encountered.